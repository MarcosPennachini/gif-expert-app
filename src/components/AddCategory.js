import React, {useState} from 'react';
import PropTypes from 'prop-types'

const AddCategory = ({categories, setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2){
            setCategories( [...categories, inputValue] );
            setInputValue('');
        }
        
    }

    const handleChange = (e) => {
        setInputValue(e.target.value);
    }
    return ( 
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder="Agrega una categorÃ­a"
                value={inputValue}
                onChange={ handleChange }
            >
            </input>
        </form>
     );
}
 
export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
