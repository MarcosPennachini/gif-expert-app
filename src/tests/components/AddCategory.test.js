import React from 'react'
import AddCategory from '../../components/AddCategory';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';

describe('Pruebas sobre componente <AddCategory />', () => {
    test('Componente <AddCategory se renderiza correctamente />', () => {
        const wrapper = shallow(<AddCategory category="categoria" setCategories={()=>{}} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})