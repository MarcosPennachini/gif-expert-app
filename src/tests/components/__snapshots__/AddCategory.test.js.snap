// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre componente <AddCategory /> Componente <AddCategory se renderiza correctamente /> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Agrega una categorÃ­a"
    type="text"
    value=""
  />
</form>
`;
